godot-escoria/escoria-demo-game
# fix: Fixes when ESCPlayer is not selectable or interactive manage click through (reopen)

PR #609 was merged a bit too fast/

Supersedes https://github.com/godot-escoria/escoria-demo-game/pull/605
Fixes https://github.com/godot-escoria/escoria-issues/issues/104

This change verifies the content of the hover stack before performing the action.

Adds a new test room 18 that allows testing superposed ESCItems.

If the clicked item is an unselectable ESCPlayer then we consider the action went through it and applies to:

    the item behind (if any in the hover stack)
    on the background if no item is behind

To test:

1. ESCPlayer superposed to ESCItem
    set mark as NOT selectable (uncheck in the inspector)
    run room 1
    hover mark should do nothing
    left/right/double click through mark
    put mark in front of and item on the wall and do the same (left/right/double click on the item through Mark)

2. Superposed ESCItems
    run room18
